tools/upgrader/util/upgrade.go

```diff
@@ -40,9 +40,9 @@ func initClient(kubeconfig string) error {
 	if kubeconfig == "" {
 		kubeconfig = util.GetDefaultKubeConfigPath()
 	}
-	config, err := clientcmd.BuildConfigFromFlags("", kubeconfig)
+	config, err := clientcmd.NewRestConfigForDefaultSearchPath(kubeconfig)
```

Comment:
Had this thought while trying to justify why the create Config methods were public, I think maybe we should add a new method called:

NewClientsForDefaultSearchPath(...) and it will return both the cluster-api client and the core kubernetes client.